tests/test_seasonal_water_yield_regression.py

```diff
@@ -1260,17 +1260,221 @@ def test_local_recharge_undefined_nodata(self):
+                            (stream_mask, stream_path)]:
+            pygeoprocessing.numpy_array_to_raster(
+                array, 0, (1, -1), (1180000, 690000), project_wkt, path)
+
+        target_b_path = os.path.join(self.workspace_dir, 'b.tif')
+        target_b_sum_path = os.path.join(self.workspace_dir, 'b_sum.tif')
+
+        seasonal_water_yield_core.route_baseflow_sum(flow_dir_mfd_path, l_path,
+                                                     l_avail_path, l_sum_path,
+                                                     stream_path, target_b_path,
+                                                     target_b_sum_path)
+
+        b = pygeoprocessing.raster_to_numpy_array(target_b_path)
+        b_sum = pygeoprocessing.raster_to_numpy_array(target_b_sum_path)
+
+        # note: obtained these arrays by running `route_baseflow_sum`
+        expected_b = numpy.array([[10.5, 0.9999998, 0],
+                                  [0.1422222, 2.2666667, 0]])
+        expected_b_sum = numpy.array([[16.916666, 1.8666663, 0],
+                                      [0.1422222, 2.5333333, 0]])
+
+        assert numpy.allclose(b, expected_b, equal_nan=True), \
+            f"Baseflow raster values do not match. Expected: {expected_b}, Got: {b}"
+        assert numpy.allclose(b_sum, expected_b_sum, equal_nan=True), \
+            f"b_sum raster values do not match. Expected: {expected_b_sum}, Got: {b_sum}"
+
+    def test_calculate_curve_number_raster(self):
+        """test `_calculate_curve_number_raster`"""
+        from natcap.invest.seasonal_water_yield import seasonal_water_yield
+        import pandas
+
+        # make small lulc raster
+        lulc_raster_path = os.path.join(self.workspace_dir, 'lulc.tif')
+        lulc_array = numpy.zeros((3, 3), dtype=numpy.int16)
+        lulc_array[1:, :] = 1
+        lulc_array[0, 0] = 2
+        make_raster_from_array(lulc_array, lulc_raster_path)
+
+        # make small soil raster
+        soil_group_path = os.path.join(self.workspace_dir, "soil_group.tif")
+        soil_groups = 4
+        soil_array = numpy.zeros((3, 3), dtype=numpy.int32)
+        for i, row in enumerate(soil_array):
+            row[:] = i % soil_groups + 1
+        make_raster_from_array(soil_array, soil_group_path)
+
+        # make biophysical table
+        biophysical_df = pandas.DataFrame([
+            {"lucode": 0, "Description": "lulc 1", "cn_a": 50,
+             "cn_b": 60, "cn_c": 0, "cn_d": 0},
+            {"lucode": 1, "Description": "lulc 2", "cn_a": 72,
+             "cn_b": 82, "cn_c": 0, "cn_d": 0},
+            {"lucode": 2, "Description": "lulc 3", "cn_a": 65,
+             "cn_b": 22, "cn_c": 1, "cn_d": 0}])
+
+        cn_path = os.path.join(self.workspace_dir, "cn.tif")
+
+        seasonal_water_yield._calculate_curve_number_raster(
+            lulc_raster_path, soil_group_path, biophysical_df, cn_path)
+
+        cn = pygeoprocessing.raster_to_numpy_array(cn_path)

```

Comment:
Sometimes it can be nice to call this variable `actual_cn` so it's easy to see that it is the data we are testing.